src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/GetAzureRmSqlDatabaseLongTermRetentionBackup.cs

```diff
@@ -0,0 +1,319 @@
+            ValueFromPipelineByPropertyName = true,
+            Position = 4,
+            HelpMessage = "The state of the database whose backups you want to find, Alive, Deleted, or All. Defaults to All")]
+        [Parameter(Mandatory = false,
+            ParameterSetName = GetBackupsByInputObjectSet,
+            Position = 2,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The state of the database whose backups you want to find, Alive, Deleted, or All. Defaults to All")]
+        [Parameter(Mandatory = false,
+            ParameterSetName = GetBackupsByResourceIdSet,
+            Position = 2,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The state of the database whose backups you want to find, Alive, Deleted, or All. Defaults to All")]
+        [ValidateNotNullOrEmpty]
+        [ValidateSet(new[] { Management.Sql.Models.DatabaseState.All, Management.Sql.Models.DatabaseState.Deleted, Management.Sql.Models.DatabaseState.Live },
+            IgnoreCase = true)]
+        public string DatabaseState { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Database object to get backups for.
+        /// </summary>
+        [Parameter(ParameterSetName = GetBackupByInputObjectSet,
+            Mandatory = true,
+            Position = 0,
+            ValueFromPipeline = true,
+            HelpMessage = "The Database object to get backups for.")]
+        [Parameter(ParameterSetName = GetBackupsByInputObjectSet,
+            Mandatory = true,
+            Position = 0,
+            ValueFromPipeline = true,
+            HelpMessage = "The Database object to get backups for.")]
+        [ValidateNotNullOrEmpty]
+        public AzureSqlDatabaseModel InputObject { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Database Resource ID to get backups for.
+        /// </summary>
+        [Parameter(ParameterSetName = GetBackupByResourceIdSet,
+            Mandatory = true,
+            Position = 0,
+            ValueFromPipeline = true,
+            HelpMessage = "The Database Resource ID to get backups for.")]
+        [Parameter(ParameterSetName = GetBackupsByResourceIdSet,
+            Mandatory = true,
+            Position = 0,
+            ValueFromPipeline = true,
+            HelpMessage = "The Database Resource ID to get backups for.")]
+        public string ResourceId { get; set; }
+
+        /// <summary>
+        /// Get the entities from the service
+        /// </summary>
+        /// <returns>The list of entities</returns>
+        protected override IEnumerable<AzureSqlDatabaseLongTermRetentionBackupModel> GetEntity()
+        {
+            if (InputObject != null)
+            {
+                LocationName = InputObject.Location;
+                ServerName = InputObject.ServerName;
+                DatabaseName = InputObject.DatabaseName;
+            }
+            else if (!string.IsNullOrWhiteSpace(ResourceId))
```

Comment:
Instead of this, use ResourceIdentifier: https://github.com/Azure/azure-powershell/blob/9ee1941722db5a5d4d1bf8fa3630184389de08b3/src/ResourceManager/KeyVault/Commands.KeyVault/Models/PSVaultIdentityItem.cs#L31

Reply:
Done.